language/move-core/types/src/state.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cell::RefCell;
+
+#[derive(Clone, Copy, Debug)]
+pub enum VMState {
+    DESERIALIZER,
+    VERIFIER,
+    RUNTIME,
+    OTHER,
```

Comment:
do we have a better name for this? not a big deal but this really means we are not within a `catch_unwind` right?